src/router.ts

```diff
@@ -13,11 +13,30 @@ import Models from "./components/Models.vue";
 Vue.use(Router);
 
 const authGuard = (to, from, next) => {
-  if (sessionStore.state.isAuthenticated) {
-    next();
+  // Check if role was loaded from local storage. This happens before App
+  // creation.
+  if (sessionStore.state.userRole === null) {
+    // Watch the role value for changes so we know when loading has happened.
+    const unwatch = store.watch(
+      () => sessionStore.state.userRole,
+      role => {
+        unwatch();
+        if (sessionStore.state.isAuthenticated) {
+          next();
+        } else {
+          store.commit("setUnauthorizedError", to.path);
+          next({ name: "home" });
+        }
+      }
+    );
   } else {
-    store.commit("setUnauthorizedError", to.path);
-    next({ name: "home" });
+    // Loading has happened, we proceed normally.
+    if (sessionStore.state.isAuthenticated) {
+      next();
+    } else {
+      store.commit("setUnauthorizedError", to.path);
+      next({ name: "home" });
+    }
```

Comment:
Consider de-duplicating the lines in this block by defining a function that you call both from the watcher above and this else clause (not relevant if we refactor to avoid asynchronicity).